core/math/quat.h

```diff
@@ -71,6 +71,37 @@ class _NO_DISCARD_CLASS_ Quat {
 		r_axis.z = z * r;
 	}
 
+	bool set_shortest_arc(const Vector3 &p_from, const Vector3 &p_to) {
+#ifdef MATH_CHECKS
+		ERR_FAIL_COND_V_MSG(false, p_from.is_zero_approx() || p_to.is_zero_approx(), "The vectors must not be zero.");
+#endif
+#ifdef REAL_T_IS_DOUBLE
+		constexpr real_t ALMOST_ONE = 0.999999999999999;
+#else
+		constexpr real_t ALMOST_ONE = 0.99999975f;
+#endif
+		Vector3 n0 = p_from.normalized();
+		Vector3 n1 = p_to.normalized();
+
+		real_t d = n0.dot(n1);
+		if (Math::abs(d) > ALMOST_ONE) {
+			return d >= 0; // Vectors are same or opposite.
```

Comment:
If the `Vector3`s are in the same direction, it returns `true`, the only thing maybe missing here is it doesn't actually set the `Quat` to identity. This isn't necessary in the case of the constructor, but probably should be done for the function call.